src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryMapping.cs

```diff
@@ -29,6 +29,8 @@ public sealed class DynamicQueryMapping
         public bool IsGroupBy { get; private set; }
 
         public List<Index> SupersededIndexes;
+        
+        public SearchEngineType SearchEngineType { get; private set; }
```

Comment:
Why do we keep info about search engine instead of `bool HasVectorSearch`? The query itself shouldn't be depended on the engine type. It's the other way around, it has a property which later on is considered for choosing the engine type that it might be executed on.

Reply:
Then you would have to keep that info per each index, IMO search engine is a good choice

Reply:
It's rather an indicator what subset of indexes can be used for particular query, so maybe we could rename it to something like `CompatibleIndexingEngineType`?